### PR TITLE
Support exporting binary gltf with separate resources.

### PR DESCRIPTION
**Pull request by JoshuaStorm**
As discussed with @lilleyse, I've added code to allow exporting non-embedded .glb files. 
There does not seem to be an issue open for this, nor do I know why anyone would ever want to write a glb with non-embedded resources, but at the very least this removes any possible ambiguity with our command line flags.

Edit: This is, however, noted in the roadmap #1

### CODE REVIEW

**Comment by pjcozzi** [comment]
> nor do I know why anyone would ever want to write a glb with non-embedded resources

The use case is keeping textures separate, see https://github.com/KhronosGroup/glTF/tree/master/extensions/Khronos/KHR_binary_glTF#performance-results

File: specs/lib/writeBinaryGltfSpec.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ describe('writeBinaryGltf', function() {
     });
 
     it('writes a valid binary gltf header', function() {
-        writeBinaryGltf(clone(testData.gltf), outputPath, true, function(err, header, scene, body) {
+        writeBinaryGltf(clone(testData.gltf), outputPath, true, true, true, function(err, header, scene, body) {
```
Comment (pjcozzi) [inline specs/lib/writeBinaryGltfSpec.js]: Add tests for when `embed` and `embedImage` are `false`.

File: lib/writeBinaryGltf.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 var path = require('path');
+var async = require('async');
 var mkdirp = require('mkdirp');
 var getBinaryGltf = require('./getBinaryGltf');
+var writeSource = require('./writeSource');
 
 module.exports = writeBinaryGltf;
 
-function writeBinaryGltf(gltf, outputPath, createDirectory, callback) {
+function writeBinaryGltf(gltf, outputPath, embed, embedImage, createDirectory, callback) {
```
Comment (pjcozzi) [inline lib/writeBinaryGltf.js]: This is starting to get unwieldy enough that I suggest changing this to take `options`.

See https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/CodingGuide#options-parameters

**Comment by pjcozzi** [comment]
@JoshuaStorm can you also post here the arguments you used for testing the command-line tool?

**Comment by JoshuaStorm** [comment]
@pjcozzi 
Command line arguments for testing:
 

Output all resources separate

`./bin/gltf-pipeline.js ./specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTest.gltf -t -s -b -o output/binaryTest.glb`

Output just image resource separately

`./bin/gltf-pipeline.js ./specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTest.gltf -t -b -o output/binaryTest.glb`

I tested a few different models and all seem to work fine. The model inspector doesn't seem to work well with nonembedded files of any type, so I tested them all in Cesium Sandcastle and they worked fine.

**Comment by pjcozzi** [comment]
Thanks @JoshuaStorm!  @lasalvavida can you do final testing and merge?

File: lib/writeBinaryGltf.js [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
 
 module.exports = writeBinaryGltf;
 
-function writeBinaryGltf(gltf, outputPath, createDirectory, callback) {
-    if (!defined(outputPath)) {
-        throw new DeveloperError('Output path is undefined.');
-    }
-    
+function writeBinaryGltf(options, callback) {
+    var gltf = options.gltf;
```
Comment (lasalvavida) [inline lib/writeBinaryGltf.js]: Leave gltf as the first parameter, don't wrap it in options.

**Comment by lasalvavida** [comment]
@JoshuaStorm There are errors after merging with master. Could you please take a look at those?

**Comment by JoshuaStorm** [comment]
@lasalvavida Resolved!

**Comment by lasalvavida** [comment]
@JoshuaStorm running a .glb through the pipeline on this branch throws an exception:

> "C:\Program Files (x86)\JetBrains\WebStorm 2016.1\bin\runnerw.exe" "C:\Program Files\nodejs\node.exe" ./bin/gltf-pipeline.js -i input/CesiumBalloon.glb

> c:\Users\rtaglang\AGI\git\gltf-pipeline\lib\writeGltf.js:21
        throw new DeveloperError('Invalid output path extension.');
        ^
> Error
    at new DeveloperError (c:\Users\rtaglang\AGI\git\gltf-pipeline\node_modules\cesium\Source\Core\DeveloperError.js:44:19)
    at writeGltf (c:\Users\rtaglang\AGI\git\gltf-pipeline\lib\writeGltf.js:21:15)
    at writeFile (c:\Users\rtaglang\AGI\git\gltf-pipeline\lib\gltfPipeline.js:114:9)
    at c:\Users\rtaglang\AGI\git\gltf-pipeline\lib\gltfPipeline.js:133:13
    at processJSONWithExtras (c:\Users\rtaglang\AGI\git\gltf-pipeline\lib\gltfPipeline.js:85:9)
    at c:\Users\rtaglang\AGI\git\gltf-pipeline\lib\gltfPipeline.js:132:9
    at c:\Users\rtaglang\AGI\git\gltf-pipeline\lib\readGltf.js:45:13
    at c:\Users\rtaglang\AGI\git\gltf-pipeline\lib\loadGltfUris.js:36:9
    at c:\Users\rtaglang\AGI\git\gltf-pipeline\node_modules\async\lib\async.js:52:16
    at done (c:\Users\rtaglang\AGI\git\gltf-pipeline\node_modules\async\lib\async.js:246:17)

**Comment by JoshuaStorm** [comment]
@lasalvavida 
Seems to be fine now, sorry about that the I missed a line in the merge fix.

**Comment by lasalvavida** [comment]
Looks good, thanks @JoshuaStorm!

**Pull request merged**